maven/WFLY-14934_Server_Plugin.adoc

```diff
@@ -0,0 +1,155 @@
+* [ ] Engineering
+
+* [X] QE
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly-extras/wildfly-jar-maven-plugin/[Bootable JAR Maven plugin]
+
+* https://github.com/eclipse/jkube[JKube]
+
+* https://github.com/wildfly/wildfly-s2i[WildFly s2i]
+
+* https://github.com/wildfly/quickstart[WildFly quickstarts]
+
+* ODO dev files.
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [ ] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Maven-controlled standalone server
+
+* [ ] Bootable jar
+
+== Requirements
+
+Ability to provision and configure a server installation.
+
+=== Hard Requirements
+
+==== Plugin requirements
+
+* Plugin produces inside the project target dir a directory (named server by default) that contains the server and any deployment.
+* By default the server is generated in ```target/server``` directory. The plugin offers a configuration item to provide the name of a directory relative to the project 
+   ```target``` directory in which the server will be installed. E.g.: ```<output-dir-name>foo-server</output-dir-name>```
+
+==== Plugin goals
+
+* ```wildfly-server:help```: Display plugin help
+* ```wildfly-server:package```: Build the application and server.
+
+==== Plugin Configuration items
+
+The detailed configuration can be found in the plugin https://github.com/jfdenise/wildfly-server-maven-plugin/wiki[community documentation]
+
+Plugin configuration items overview:
+
+* cli-sessions: List of CLI execution sessions
+* context-root: To make the WAR deployment registered under root resource path ('/').
+* display-cli-scripts-output: By default executed CLI scripts output is not shown if execution
+* excluded-layers: A list of Galleon layers to exclude
+* extra-server-content-dirs: A list of directories to copy content to the provisioned server
+* feature-pack-location: The WildFly Galleon feature-pack location
+* feature-packs: A list of feature-pack configurations to install
+* hollow-server: Create a server that doesn’t contain application
+* layers: A list of Galleon layers to provision.
+* log-time: Whether to log provisioning time at the end
+* off-line: Whether to use offline mode when the plugin resolves an artifact
+* output-dir-name: By default the generated server is 'server'
+* plugin-options: Arbitrary Galleon options used when provisioning the server
+* project-build-dir: Project build dir.
+* provisioning-file: The path to the provisioning.xml file to use
+* record-state: Whether to record provisioned state in .galleon directory.
+* skip: To skip execution
+
+==== Plugin artifacts
+
+===== WildFly Server Maven Plugin
+
+The maven plugin GAV is ```org.wildfly.plugins:wildfly-server-maven-plugin:1.0.0.Final```.
+The source repository is expected to be ```http://github.com/wildfly-extras/wildfly-server-plugin```
```

Comment:
should it be `http://github.com/wildfly-extras/wildfly-server-maven-plugin` to be consistent?

Reply:
Typo, that is http://github.com/wildfly-extras/wildfly-server-maven-plugin